napari/viewer.py

```diff
@@ -45,7 +44,24 @@ def __init__(
             order=order,
             axis_labels=axis_labels,
         )
-        self.window = Window(self, show=show)
+
+        # TODO: discuss.  This is THE entrypoint for Qt stuff here.
+        try:
+            from ._qt import Window
+
+            self.window = Window(self, show=show)
+        except ImportError:
+
+            class Window:
+                def close(self):
+                    pass
+
+                def __getattr__(self, name):
+                    raise ImportError(
+                        "could not import qtpy.  Cannot show napari window."
+                    )
+
```

Comment:
> Similarly I've modified the from ._qt import Window in napari/viewer.py to create a mock Window object that will only raise an exception if someone tries to access anything on viewer.window. Could move that to a dedicated file, or modify as requested.

I think I'd rather have this in a dedicated file to try and keep `Viewer` as clean and simple as possible.

Reply:
i create a `components.window` module. let me know how that strikes you.  I think it could be a place, in the future, where we define what a graphical backend needs to implement to work with napari.  (something like the [widget protocol in magicgui](https://github.com/napari/magicgui/blob/master/magicgui/widgets/_protocols.py))

Reply:
File looks good and I like the idea, but I think i'd rather it go at the top level instead of in `components`, I don't really like having to go `from .._qt import Window` inside `components` and could one day see the whole components move to a `napari-models` folder which would be entirely independent from all our backends. Thoughts?

Reply:
top level as in `napari.window`?  that's fine with me too

Reply:
An alternative view would be that the `components` folder would stay inside `napari`, there would be no `napari-models`, but instead all the qt/ vispy stuff would go to `napari-qt` and then we'd have `napari-electron` or something else, that you could have behind `napari`. We don't need to decide this now, and we probably want a top level window import regardless ...

Reply:
yeah, as I just noted, i can see arguments for the other direction too, but I'd probably go with `napari.window` for now